expression/builtin_op.go

```diff
@@ -799,7 +799,12 @@ func (c *isNullFunctionClass) getFunction(ctx sessionctx.Context, args []Express
 		sig = &builtinRealIsNullSig{bf}
 		sig.setPbCode(tipb.ScalarFuncSig_RealIsNull)
 	case types.ETDatetime:
-		sig = &builtinTimeIsNullSig{bf}
+		isNotNull := false
+		_, isCol := args[0].(*Column)
+		if isCol && mysql.HasNotNullFlag(args[0].GetType().Flag) {
```

Comment:
```suggestion
		if isCol && mysql.HasNotNullFlag(col.GetType().Flag) {
```
There is no need to index the slice again.